src/Ookii.Dialogs.Wpf/ProgressDialog.cs

```diff
@@ -103,6 +124,15 @@ public ProgressDialog(IContainer container)
             // Set a default animation for XP.
             if( !NativeMethods.IsWindowsVistaOrLater )
                 Animation = AnimationResource.GetShellAnimation(Ookii.Dialogs.Wpf.ShellAnimation.FlyingPapers);
+
+            // Set the default cancellation Token source.
+            _cancellationSource = new System.Threading.CancellationTokenSource();
+
+            // Link with incoming Token if required.
+            if (!token.Equals(System.Threading.CancellationToken.None))
+            {
+                _cancellationLinked = System.Threading.CancellationTokenSource.CreateLinkedTokenSource(token, _cancellationSource.Token);
```

Comment:
I think it's safe to have only one `CancellationTokenSource` and always create a linked token source, even when the `CancellationToken` provided is `CancellationToken.None` (i.e. it works fine)

Reply:
CancellationTokenSources are disposable, so at least you need to track both the internally created one, and the one used for linking, but it's fine to always create the linked one

Reply:
If we always create a single `CancellationTokenSource` that is linked to the `CancellationToken` provided, we own that `CancellationTokenSource` and can safely dispose it after the background worker finishes - disposing our `CancellationTokenSource` will have no effect out the original (source) `CancellationToken` which is good